src/lib/types/events.ts

```diff
@@ -1333,19 +1333,27 @@ export class UserUpdatedEvent extends BaseEvent {
 }
 
 export class UserDeletedEvent extends BaseEvent {
-    readonly preData: IUserWithRootRole;
+    readonly preData: IUserEventData;
 
     constructor(eventData: {
         createdBy: string | IUser;
-        deletedUser: IUserWithRootRole;
+        deletedUser: IUserEventData;
         createdByUserId: number;
     }) {
         super(USER_DELETED, eventData.createdBy, eventData.createdByUserId);
         this.preData = mapUserToData(eventData.deletedUser);
     }
 }
 
-function mapUserToData(user: IUserWithRootRole): any {
+interface IUserEventData extends Partial<IUserWithRootRole> {
+    id: number;
+    name: string;
+    username: string;
+    email: string;
```

Comment:
- name vs username, what's the difference?
- email is sometimes optional, I believe the username might be as well, so the type I have in mind is something like `Either<UserWithUsername, UserWithEmail>`

Reply:
> name vs username, what's the difference?

Good question. No idea. It depends on how the user got into the system and when they got into the system. Legacy flows are slightly different. SSO is slightly different (also depending on provider and setup, you might get name but not username or vice versa). 

I don't believe there _is_ a meaningful difference, only us doing terrible things to the types 

> email is sometimes optional, I believe the username might be as well, so the type I have in mind is something like Either<UserWithUsername, UserWithEmail>

We spoke already. This is for anyone trying to find bread crumbs at a future date (hello future person!). This doesn't work. There's cases where none of these fields are set, even thought the type system says they should be